killbill-integration-tests/mixin-checkers/invoice_checker.rb

```diff
@@ -26,6 +33,21 @@ def check_invoice_item(ii, invoice_id, amount, currency, item_type, plan_name, p
       assert_equal(end_date, ii.end_date, msg)
     end
 
+    def check_invoice_item_detail(ii, usage_input, amount)
+      msg = "invoice_item #{ii.invoice_item_id}"
+      assert_not_nil(ii.item_details, msg)
+      item_details = JSON.parse(ii.item_details, :symbolize_names => true )
+      detail_amount = 0
+      item_details.each_with_index do |item_detail, index|
+        assert_equal(usage_input[index][:unit_type], item_detail[:tierUnit], msg)
```

Comment:
@pierre, for `consumable` how the tier number will be based, by `tiered block` or by `tier`?

Reply:
It would be by `tier`.

Using https://github.com/killbill/killbill/issues/839#issuecomment-362474537 as an example (`CONSUMABLE` section),

* Case 1: `FOO` and `BAR` are both in tier 1
* Case 2: `FOO` in tier but `BAR` across tier 1 (100*2) and tier 2 (1 * 20)
* Case 3: `FOO` across all 3 tiers and `BAR` across tier 1 and tier 2